### PR TITLE
Adopt retrieval of application packages during staging

### PR DESCRIPTION
**Pull request by enchobelezirev**
#### Description: 
When the upload of the application is not skipped and the application is not restarted because there are no changes in the content - a problem might rise if the application is restaged, after the validation phase of the blue green deployment.

### CODE REVIEW

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StageAppStep.java [hunk 1]
```diff
@@ -26,8 +27,11 @@
 
     @Override
     protected StepPhase executeAsyncStep(ExecutionWrapper execution) {
-        CloudApplication app = StepsUtil.getApp(execution.getContext());
-        ApplicationStager applicationStager = new ApplicationStager(execution.getControllerClient());
+        String appName = StepsUtil.getApp(execution.getContext())
+                                  .getName();
+        CloudControllerClient client = execution.getControllerClient();
+        CloudApplication app = client.getApplication(appName);
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StageAppStep.java]: Can we avoid the fetch of this app?

File: com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/util/ApplicationStagerTest.java [hunk 2]
```diff
@@ -214,10 +218,49 @@ public void testBindDropletToApp() {
...
+               .thenReturn(null);
+        UUID packageGuid = UUID.randomUUID();
+        UUID testApplicationGuid = UUID.randomUUID();
+        CloudApplication testApplication = ImmutableCloudApplication.builder()
+                                                                    .metadata(ImmutableCloudMetadata.builder()
+                                                                                                    .guid(testApplicationGuid)
+                                                                                                    .build())
+                                                                    .name("test-app")
+                                                                    .build();
+        CloudPackage andAndOnlyApplicationPackage = ImmutableCloudPackage.builder()
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/util/ApplicationStagerTest.java]: There are two `and`s in this variable name.

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ApplicationStager.java [hunk 3]
```diff
@@ -133,11 +137,28 @@ public void bindDropletToApplication(DelegateExecution context, UUID appGuid) {
 
     public StepPhase stageApp(DelegateExecution context, CloudApplication app, StepLogger stepLogger) {
         UploadToken uploadToken = StepsUtil.getUploadToken(context);
-        if (uploadToken == null) {
-            return StepPhase.DONE;
+        if (uploadToken != null) {
+            stepLogger.debug(Messages.UPLOAD_TOKEN_FOR_APPLICATION_0_1, app.getName(), JsonUtil.toJson(uploadToken, true));
+            return stageApplication(context, app, stepLogger, uploadToken);
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ApplicationStager.java]: Do `uploadToken.getPackageGuid()` here and remove the `stageApplication` method overload below.

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StageAppStep.java (hunk 1 above)
Comment (enchobelezirev) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StageAppStep.java]: We need the GUID...
It is in the ApplicationStager app.getMetadata().getGuid()

**Pull request merged**